src/utils.h

```diff
@@ -79,25 +83,31 @@ extern gboolean use_syslog;
 	do { \
 		fprintf(stderr, "[conmon:e]: " fmt "\n", ##__VA_ARGS__); \
 		if (use_syslog) \
-			syslog(LOG_ERR, "conmon %.20s <error>: " fmt " \n", log_cid, ##__VA_ARGS__); \
+			syslog(LOG_ERR, "conmon %.20s <error>: " fmt "\n", log_cid, ##__VA_ARGS__); \
 		exit(EXIT_FAILURE); \
 	} while (0)
 
 #define pwarn(s) \
 	do { \
-		fprintf(stderr, "[conmon:w]: %s %s\n", s, strerror(errno)); \
-		if (use_syslog) \
-			syslog(LOG_INFO, "conmon %.20s <pwarn>: %s %s\n", log_cid, s, strerror(errno)); \
+		if (log_level >= WARN_LEVEL) { \
+			int saved_errno = errno; \
+			fprintf(stderr, "[conmon:w]: %s %s\n", s, strerror(saved_errno)); \
+			if (use_syslog) \
+				syslog(LOG_INFO, "conmon %.20s <pwarn>: %s %s\n", log_cid, s, strerror(saved_errno)); \
+			errno = saved_errno; \
```

Comment:
The restores seem unnecessary? (2 cases)

Reply:
Yes, I can remove these as it seems an overkill. I'd need to add a similar thing to all debug() and info() macros if we want to go ahead with a paradigm that logging functions themselves shouldn't affect errno.

Reply:
I generally think that the most reliable implementation approach is: if anything needs an `errno` value, it _must_ capture it immediately after a failing call itself; a rule that “some functions preserve errno” is too difficult to maintain, and tools won’t help us.